ib_manifest_util/util.py

```diff
@@ -136,3 +142,83 @@ def download_files(urls: str | list) -> list:
         logger.info(f"Downloaded file {filename} from {address}.")
 
     return filenames
+
+
+def verify_local_channel_environments(
+    local_channel_env: str | Path,
+    offline_mode: bool = True,
+    conda_binary_loc: str | Path | None = None,
+):
+    """
+    Verify that the conda environment (`local_channel_env`) can be successfully created.
+
+    Args:
+        local_channel_env: path to local environment yaml
+        offline_mode: whether to run `conda env create` with the `--offline` flag, default is true
+        conda_binary_loc: path to a `conda` binary, default will try to locate the `conda` binary by using the return value of `$ which conda`
+
+    NOTE: It is assumed that the `conda-vendor vendor` command has already been run and that the `local_channel_env`
+    has a single channel set to `path/to/local_channel/folder`.
+    """
+
+    logger.info(
+        "Verifying that the local channel environment can be successfully created"
+    )
+    if isinstance(local_channel_env, str):
+        local_channel_env = Path(local_channel_env)
+
+    if not local_channel_env.exists():
+        raise ValueError(
+            f"The environment file submitted appears not to exist at location: {local_channel_env.resolve()}"
+        )
+
+    env_yaml = load_yaml(local_channel_env)
+    channels = env_yaml["channels"]
+    name = env_yaml["name"]
```

Comment:
```suggestion
    channels = env_yaml.get("channels", None)
    name = env_yaml.get("name", None)
```
This won't error if the fields are not present